vault/identity_store_oidc.go

```diff
@@ -1085,10 +1094,37 @@ func (k *namedKey) rotate(ctx context.Context, s logical.Storage, overrideVerifi
 
 // generateKeys returns a signingKey and publicKey pair
 func generateKeys(algorithm string) (*jose.JSONWebKey, error) {
-	// 2048 bits is recommended by RSA Laboratories as a minimum post 2015
-	key, err := rsa.GenerateKey(rand.Reader, 2048)
-	if err != nil {
-		return nil, err
+	var key interface{}
+	var err error
+
+	switch algorithm {
+	case "RS256", "RS384", "RS512":
+		// 2048 bits is recommended by RSA Laboratories as a minimum post 2015
+		if key, err = rsa.GenerateKey(rand.Reader, 2048); err != nil {
```

Comment:
Could be an item to look at in the future, but I think there was some discussion about using something other than 2048 or making this number configurable.

Reply:
I think this can be made configurable later, although I also think it's easy to add a key_bits parameter, so...looking for something to do? :-)

Reply:
I left it be for right now since it isn't immediately required and we can discuss more generally. When I did a little research into it, there was a lot of push to upgrade keys by moving to EC or EdDSA, vs. larger RSA keys. It can definitely be added pretty easily if we choose to or there is demand.